### PR TITLE
📖  update documentation to push container images using docker buildx

### PR DESCRIPTION
**Pull request by ashish-amarnath**
Signed-off-by: Ashish Amarnath <ashisham@vmware.com>

For the benefit of reviewers: https://deploy-preview-2854--velero.netlify.app/docs/main/build-from-source/

### CODE REVIEW

**Comment by carlisia** [comment]
Awesome, this works! Thanks so much for putting the instruction together, and I love the inclusion of the wrong/right outputs.

I have only a suggestion for improving formatting:
image

For (1), I'd tab left in a way that it'd be lined up with the `2.` above it. Because it's a new section. Otherwise it looks like a continuation of that `2.` section. And remove the `:`. For (2), just add `:` in both those places.

**Comment by nrb** [comment]
I think the formatting is a result of our Markdown renderer; it may be tough to move the NOTE out to the left without fighting against the renderer's rules, though I do see your point.

I'm trying to see if there's an equivalent to [admonitions](https://docutils.sourceforge.io/docs/ref/rst/directives.html#admonitions), but so far I'm turning up empty for them in Markdown.

**Comment by nrb** [comment]
In use, an `admonition` directive would look something like this warning box (It's actually a `warning` admonition)

Screen Shot 2020-08-25 at 3 54 37 PM

**Comment by ashish-amarnath** [comment]
Yeah, I am trying a plethora of combinations to get the formatting right, unsuccessfully though.

**Comment by ashish-amarnath** [comment]
The equivalent formatting for what Nolan pasted was the intention

File: site/content/docs/main/build-from-source.md [hunk 1]
```diff
@@ -98,6 +98,69 @@ make container
...
+    Name:   default
+    Driver: docker
+
+    Nodes:
+    Name:      default
+    Endpoint:  default
+    Status:    running
+    Platforms: linux/amd64, linux/arm64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
+    ```
+    And the `REGISTRY=myrepo BUILDX_OUTPUT_TYPE=registry make container` will fail withthe below error
```
Comment (nrb) [inline site/content/docs/main/build-from-source.md:141]: `withthe` => `with the`

File: site/content/docs/main/build-from-source.md [hunk 2]
```diff
@@ -98,6 +98,69 @@ make container
...
+    Name:   builder
+    Driver: docker-container
+
+    Nodes:
+    Name:      builder0
+    Endpoint:  unix:///var/run/docker.sock
+    Status:    running
+    Platforms: linux/amd64, linux/arm64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
+    ```
+    NOTE:
```
Comment (nrb) [inline site/content/docs/main/build-from-source.md:127]: Can you put the actual note after this `NOTE:`?

File: site/content/docs/main/build-from-source.md [hunk 3]
```diff
@@ -98,6 +98,69 @@ make container
...
+    Platforms: linux/amd64, linux/arm64, linux/ppc64le, linux/s390x, linux/386, linux/arm/v7, linux/arm/v6
+    ```
+    And the `REGISTRY=myrepo BUILDX_OUTPUT_TYPE=registry make container` will fail withthe below error
+    ```bash
+    $ REGISTRY=ashishamarnath BUILDX_PLATFORMS=linux/arm64 BUILDX_OUTPUT_TYPE=registry make container
+    auto-push is currently not implemented for docker driver
+    make: *** [container] Error 1
+    ```
+
+Having completed the above one time setup, now the output of `docker buildx inspect --bootstrap` should be like
```
Comment (nrb) [inline site/content/docs/main/build-from-source.md:148]: I think there should be a newline between the prose and the code blocks throughout, unless the new Markdown renderer is more generous with spacing.

File: site/content/docs/main/build-from-source.md (hunk 1 above)
Comment (ashish-amarnath) [reply site/content/docs/main/build-from-source.md:141]: spotted and fixed.

File: site/content/docs/main/build-from-source.md (hunk 3 above)
Comment (ashish-amarnath) [reply site/content/docs/main/build-from-source.md:148]: the preview looked good. But I am going to add the new lines. Better for human consumption.

**Comment by ashish-amarnath** [comment]
@carlisia and @zubron PTAL!

**Pull request merged**